ss58-js/mod.ts

```diff
@@ -0,0 +1,89 @@
+import { blake2b } from "../deps/hashes.ts";
+import * as base58 from "../deps/std/encoding/base58.ts";
+
+import { hex } from "../util/mod.ts";
+
+import {
+  ALLOWED_ADDRESS_LENGTHS,
+  ALLOWED_NETWORK_PREFIXES,
+  ALLOWED_PUBLIC_KEY_LENGTHS,
+  CHECKSUM_LENGTH,
+  SS58PRE,
+} from "./constants.ts";
+
+export const encode = (prefix: number, pubKey: string): string => {
```

Comment:
Let's change the signature

```diff
- (prefix: number, pubKey: string): string;
+ (prefix: number, pubKey: Uint8Array): Uint8Array;
```

Line 15 can then be deleted.

Reply:
Agreed – the raw logic should be decoupled from the string-encoding, so that raw `Uint8Array`s can be passed around internally

Reply:
@harrysolovay @tjjfvi  if there is not need to encode it into a string, then, the base68 dependency is not needed for this, right?

Reply:
I'm a bit confused on the suggested signatures for encode/decode
```
export const encode = (prefix: number, pubKey: Uint8Array): Uint8Array
export const decode = (address: Uint8Array): [prefix: number, pubKey: Uint8Array]
```

if `encode` returns bytes or `decode` receive bytes, base58 encoding/decoding is not used.

Note: I'm assuming that the SS58 address format needs to be base58 encoded as described [here](https://github.com/substrate-developer-hub/substrate-docs/blob/archive/main-mdx/v3/docs/07-advanced/h-ss58/index.mdx#basic-format) 
```
base58encode ( concat ( <address-type>, <address>, <checksum> ) )
```

Reply:
I think we should rename `encode`/`decode` to `encodeRaw`/`decodeRaw` and add
```ts
export const encode = (prefix: number, pubKey: Uint8Array) => base58.encode(encodeRaw(prefix, pubkey));
export const decode = (address: string) => decodeRaw(base58.decode(address));
```

Most of the time, one will use `encode`/`decode`, but it is worthwhile to have `encodeRaw`/`decodeRaw` as options in the rare cases where one deals with the raw data.

Reply:
makes sense 😄